runtime/jcl/common/mgmtmemory.c

```diff
@@ -76,25 +76,77 @@ Java_com_ibm_java_lang_management_internal_MemoryMXBeanImpl_getNonHeapMemoryUsag
 	omrthread_monitor_enter(javaVM->classTableMutex);
 	classLoader = javaVM->internalVMFunctions->allClassLoadersStartDo(&walkState, javaVM, 0);
 	while (NULL != classLoader) {
-		UDATA *udataFreeListBlock = classLoader->ramClassUDATABlockFreeList;
-		J9RAMClassFreeListBlock *tinyFreeListBlock = classLoader->ramClassTinyBlockFreeList;
-		J9RAMClassFreeListBlock *smallFreeListBlock = classLoader->ramClassSmallBlockFreeList;
-		J9RAMClassFreeListBlock *largeFreeListBlock = classLoader->ramClassLargeBlockFreeList;
-		while (NULL != udataFreeListBlock) {
-			used -= sizeof(UDATA);
-			udataFreeListBlock = *(UDATA **) udataFreeListBlock;
+		J9RAMClassUDATABlockFreeList *udataFreeListBlock = &classLoader->ramClassUDATABlocks;
+		J9RAMClassFreeLists *sub4gFreeListBlock = &classLoader->sub4gBlock;
+		J9RAMClassFreeLists *freqFreeListBlock = &classLoader->frequentlyAccessedBlock;
+		J9RAMClassFreeLists *InFreqFreeListBlock = &classLoader->inFrequentlyAccessedBlock;
+		if (NULL != udataFreeListBlock) {
```

Comment:
This check is not necessary since `classLoader->ramClassUDATABlocks` is not a pointer. This applies to the J9RAMClassFreeLists's as well.